packages/server/src/events/tasks.ts

```diff
@@ -29,7 +31,26 @@ export async function AddDislikeActivity(email: string, articleId: number) {}
  * Task ( Runs after the article save is created successfully ).
  * Add save activity to the logged in user
  */
-export async function AddSaveActivity(email: string, articleId: number) {}
+export async function AddSaveActivity(email: string, articleId: number) {
```

Comment:
It is good approach but it will send multiple requests to the database. Better approach is this : 

```js 

export async function AddSaveActivity(email: string, articleId: number) {
    const prisma = PrismaClientSingleton.prisma;
    // add save activity to the logged in user
    await prisma.user.update({
        where: {
            email: email,
        },
        data: {
            articleActivities: {
                createMany: {
                    data: {
                        articleId: articleId,
                        action: 'save',
                    }
                }
            }
        }
    });
}

```

Reply:
You are doing great, but from now on choose this approach. Thanks